src/pybind/mgr/dashboard/frontend/src/app/core/error/error.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, HostListener, OnDestroy, OnInit } from '@angular/core';
+import { NavigationEnd, Router, RouterEvent } from '@angular/router';
+
+import { Subscription } from 'rxjs';
+import { filter } from 'rxjs/operators';
+
+import { DocService } from '~/app/shared/services/doc.service';
+
+@Component({
+  selector: 'cd-error',
+  templateUrl: './error.component.html',
+  styleUrls: ['./error.component.scss']
+})
+export class ErrorComponent implements OnDestroy, OnInit {
+  header: string;
+  message: string;
+  section: string;
+  section_info: string;
+  icon: string;
+  docUrl: string;
+  source: string;
+  routerSubscription: Subscription;
+
+  constructor(private router: Router, private docService: DocService) {}
+
+  ngOnInit() {
+    this.fetchData();
+  }
+
+  @HostListener('window:beforeunload', ['$event']) unloadHandler(event: Event) {
+    event.returnValue = false;
+  }
+
+  fetchData() {
+    try {
+      this.message = history.state.message;
+      this.header = history.state.header;
+      this.section = history.state.section;
+      this.section_info = history.state.section_info;
+      this.icon = history.state.icon;
+      this.source = history.state.source;
+      this.router.onSameUrlNavigation = 'reload';
+      this.docUrl = this.docService.urlGenerator(this.section);
+      this.routerSubscription = this.router.events
```

Comment:
I detected a recursion problem here:
Everytime ```fetchData``` is called, a new subscription is created,
so after visiting several times the error page, I ended up with tens of subscription
and my browser tab started to become unresponsive,
so I suggest to avoid recursion and/or handle unsubscriptions properly.

Reply:
Hey @alfonsomthd , I have made some modifications to get rid of this recursion issue, Can you please re-test and confirm if it still persists? Thanks.

Reply:
> Hey @alfonsomthd , I have made some modifications to get rid of this recursion issue, Can you please re-test and confirm if it still persists? Thanks.

I see you're unsubscribing each time in order to avoid the side effect of the recursion,
but (just an observation) wouldn't be better to avoid recursion instead of having to unsubscribe?

Reply:
Thanks @alfonsomthd , refactored according to your suggestion.